Optional Parameters

luise opened the pull request:
This PR makes a few changes that simplify how we show optional parameters in our docs.

## Signatures:
_After:_
<img width="330" alt="screen shot 2017-10-30 at 10 58 40 am" src="https://user-images.githubusercontent.com/10263364/32187156-58b51868-bd61-11e7-835e-015a6cff694b.png">

_Before:_
<img width="342" alt="screen shot 2017-10-30 at 10 58 48 am" src="https://user-images.githubusercontent.com/10263364/32187166-5e023d1e-bd61-11e7-9d1b-dbbb426442e7.png">


## Param Table:
_After:_
<img width="1210" alt="screen shot 2017-11-03 at 10 00 16 am" src="https://user-images.githubusercontent.com/10263364/32386271-d4b0867c-c07d-11e7-91f6-b8649e43f14e.png">


_Before:_ 
<img width="1143" alt="screen shot 2017-10-30 at 10 43 58 am" src="https://user-images.githubusercontent.com/10263364/32187123-40a57c90-bd61-11e7-9026-1e6cebf0ad94.png">

kayousterhout commented on tmpl/params.tmpl:
```diff
@@ -1,7 +1,7 @@
     var params = obj;
 
-    /* sort subparams under their parent params (like opts.classname) */
+    /* list subparams as parentParam.subparam */
```
FWIW I find the new thing harder to read, because it wouldn't be totally obvious to me what it means that "opts.Param" is shown as the parameter name

kayousterhout commented on tmpl/params.tmpl:
```diff
@@ -87,7 +87,9 @@
         <tr>
             <?js if (params.hasName) {?>
-                <td class="name" style="white-space:nowrap;"><code><?js= param.name ?></code></td>
+                <td class="name" style="white-space:nowrap;"><code><?js= param.name ?></code>
```
This is awesome and looks way better than before!

kayousterhout commented on publish.js:
```diff
@@ -108,6 +104,13 @@ function updateItemName(item) {
             attributes.join(', ') );
     }
 
+    if (index !== 0) {
```
Thanks for fixing this the "optopt" thing that sometimes resulted was super ugly

luise commented:
Thanks for the comments, @kayousterhout ! I removed the commit that got rid of the nested tables. I think the nested tables are kind of clunky, but I also see how `opts.attr` can be confusing. I updated the screenshot in the PR description to reflect the change.

kayousterhout commented on publish.js:
```diff
@@ -95,7 +91,7 @@ function getSignatureAttributes(item) {
 }
 
-function updateItemName(item) {
+function updateItemName(item, index) {
```
Does line 120 need to be updated now?

luise commented:
@kklin , ready!
